spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/h2/H2ConsoleAutoConfiguration.java

```diff
@@ -63,9 +63,10 @@
 		String urlMapping = path + (path.endsWith("/") ? "*" : "/*");
 		ServletRegistrationBean<WebServlet> registration = new ServletRegistrationBean<>(new WebServlet(), urlMapping);
 		configureH2ConsoleSettings(registration, properties.getSettings());
-		dataSource.ifAvailable((available) -> {
+		logger.info("H2 console available at '" + path + "'.");
+		dataSource.orderedStream().forEachOrdered((available) -> {
 			try (Connection connection = available.getConnection()) {
-				logger.info("H2 console available at '" + path + "'. Database available at '"
+				logger.info("Database available at '"
```

Comment:
I'd rather create a single log entry with them. Ideally, something that shortcuts to the following log if there is only one and an optimized one for the multi-datasources scenario.

Reply:
Thank you for the kind suggestion, I'll implement that. I actually this in mind as well, but did not want to overcomplicate things, this being my first contribution.

Reply:
Yeah, let's see how that will look like and we can refine if necessary. Thanks for the contribution!